### ISSUE TITLE
Add "arns" as an output to "data aws_autoscaling_groups"

### ISSUE THREAD

**Issue by Hashfyre** [terraform-providers/terraform-provider-aws#1909]
### Feature Request
Add `arns` as an output in addition to `names` in https://www.terraform.io/docs/providers/aws/d/autoscaling_groups.html

#### Example usage:

```
data "aws_autoscaling_groups" "beta" {
  filter {
    name   = "key"
    values = ["Env"]
  }

  filter {
    name   = "value"
    values = ["beta"]
  }
}

data "aws_iam_policy_document" "asg_policy" {
  statement {
    effect = "Allow"

    actions = [
      "ec2:DescribeInstances",
      "ec2:TerminateInstances",
      "autoscaling:DescribeAutoScalingInstances",
      "autoscaling:DetachInstances",
    ]

    resources = ["${data.aws_autoscaling_groups.beta.arns}"]
  }
}
```

**Comment by bflad** [comment]
The `arns` attribute has been merged into master and will release with version 1.38.0 of the AWS provider, likely middle of next week.

@blckct currently the "plural" data sources are generally implemented only to return identifiers (IDs/names/ARNs) for easily iterating through the list (e.g. creating IAM policies). With some of the enhancements coming in Terraform 0.12, it will be easier to work with full objects instead of simple attributes, so we can entertain ideas like you mention, but we would certainly be curious about the use cases. We also have some very early design ideas to combine the notions of resources and data sources. Its probably best to capture this all in a separate issue. 👍

**Comment by bflad** [comment]
This has been released in [version 1.38.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1380-september-26-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
